function.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Roger Chapman and the v8go contributors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package v8go
+
+// #include "v8go.h"
+import "C"
+import (
+	"unsafe"
+)
+
+// Function is a JavaScript function.
+type Function struct {
+	*Value
+}
+
+// Call this JavaScript function with the given arguments.
+func (fn *Function) Call(args []*Value) (*Value, error) {
+	var rtn C.RtnValue
+	if len(args) == 0 {
+		rtn = C.FunctionCall(fn.ptr, C.int(0), (*C.ValuePtr)(nil))
+	} else {
```

Comment:
Is this else statement needed? The range would still work with a empty array?
If we need to pass the nil pointer can we return in the above `if` statement to avoid the extra branching caused by the `else`

Reply:
That was something I found by writing the test -- to get the `*C.ValuePtr`, I have to take the value of the first element, and that is an illegal index on an empty slice. 

Updated to avoid calling FunctionCall in two different branches.